pkg/kubeconfig/kubeconfig.go

```diff
@@ -17,22 +17,124 @@ limitations under the License.
 package kubeconfig
 
 import (
+	"bytes"
 	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
 	"io/fs"
 	"net"
 	"os"
+	"time"
 
 	kubeoneapi "k8c.io/kubeone/pkg/apis/kubeone"
+	"k8c.io/kubeone/pkg/certificate"
 	"k8c.io/kubeone/pkg/executor"
 	"k8c.io/kubeone/pkg/executor/executorfs"
 	"k8c.io/kubeone/pkg/fail"
 	"k8c.io/kubeone/pkg/state"
 
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
+	clientapi "k8s.io/client-go/tools/clientcmd/api"
+	clientapipublic "k8s.io/client-go/tools/clientcmd/api/latest"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	pkiCAcrt = "/etc/kubernetes/pki/ca.crt"
+	pkiCAkey = "/etc/kubernetes/pki/ca.key"
+)
+
+func GenerateSuperAdmin(st *state.State, cn string, on []string, ttl time.Duration) ([]byte, error) {
```

Comment:
This function can generate any kubeconfig now, so we should rename it to something else. Maybe `GenerateKubeconfig` or just `Generate`?